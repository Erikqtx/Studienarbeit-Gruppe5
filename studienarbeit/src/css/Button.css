/*Country-State Slider Button*/

.button-box1{
    width: 20%;
    margin-left: 10px;
    margin-top: 15px;
    position: relative;
    box-shadow: 9px 2px 16px -8px rgba(0,0,0,0.5);
    border-radius: 5px;
    background: rgba(195, 216, 221, 0.301);
    transform: translate(15%, 10%);
    overflow: hidden;
    float: left;
    
  }
  
  .button-box1 label {
    padding: 3px 0 3px 0;
    width: 50%;
    float: left;
    color:white;
    cursor: pointer;
    position: relative;
    font-size: 13px;
    text-align: center;
    font-family: 'Source Sans Pro', sans-serif;
  }
  
  
  .slider{
    position:absolute;
    left: 0%; 
    top:0;
    width:50%; 
    height: 100%;
    animation-timing-function: ease-in-out;
    animation-duration:.5s;
    transition: 0.1s left .05s ease-in-out;
    z-index: -1;
  }
 
  #country + label:before{
    content: "";
    width: 100%;
    height: 100%;
    position: absolute;
    top: 0;
    left: 100%;
    z-index: -1;
    background-color: #fff;
    transition: all 0.3s ease;
  }
  
  #country:checked + label:before{
    left: 0;
  }
  
  #country:checked + label,
  #state:checked + label{
    color:black;
  }

/*Year 2015-2016 Slider Button*/

.button-box2{
  width: 20%;
  margin-right: 10%;
  margin-top: 15px;
  position: relative;
  box-shadow: 9px 2px 16px -8px rgba(0,0,0,0.5);
  border-radius: 5px;
  background: rgba(195, 216, 221, 0.301);
  transform: translate(15%, 10%);
  overflow: hidden; 
  float: right;

  
}

.button-box2 label {
    padding: 3px 0 3px 0;
    width: 50%;
    float: left;
    color:white;
    cursor: pointer;
    position: relative;
    font-size: 13px;
    text-align: center;
    font-family: 'Source Sans Pro', sans-serif;
  }

  #year1 + label:before{
    content: "";
    width: 100%;
    height: 100%;
    position: absolute;
    top: 0;
    left: 100%;
    z-index: -1;
    background-color: #fff;
    transition: all 0.3s ease;
  }
  
  #year1:checked + label:before{
    left: 0;
  }
  
  #year1:checked + label,
  #year2:checked + label{
    color:black;
  }
  

  @media only screen and (max-width: 600px) {
    
    .button-box1,
    .button-box2{
      width: 30%;
    }
    
    .button-box1 label,
    .button-box2 label {
      font-size: 12px;
    }
  
  }