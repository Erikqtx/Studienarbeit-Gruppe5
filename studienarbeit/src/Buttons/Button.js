import React from "react";
import "../css/Button.css";


function Button() {
  
   function Country(){
     console.log("Country")
   }

   function State(){
    console.log("State")
  }

  function year2015(){
    console.log("chartof2015")
  }

  function year2016(){
    console.log("chartof2016")
  }

  return (
    <div className="parentbox">
        <button data-testid="country" className ="country" onClick = {Country}>Country</button>
        <button className ="state" onClick = {State}>State</button>
        <div/>
        <button className ="year1" onClick = {year2015}>2015</button>
        <button className ="year2" onClick = {year2016}>2016</button>
    </div>
  );
}

export default Button;