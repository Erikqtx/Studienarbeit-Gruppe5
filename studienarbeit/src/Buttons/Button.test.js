import React from 'react';
import ReactDOM from 'react-dom';
import Button from "./Button";
import { render, getByTestId, fireEvent, cleanup } from "@testing-library/react";
import "@testing-library/jest-dom";
import renderer from 'react-test-renderer';

it("renders without crashing", () => {
    const div = document.createElement("div");
    ReactDOM.render(<Button/>, div)
})

it("matches the snapshot", () => {
    const tree = renderer.create(<Button/>).toJSON();
    expect(tree).toMatchSnapshot();
})

it("captures clicks", () => {
    function handleClick() {
    }
    const { getByTestId } = render(
        <Button onClick={handleClick}>Country</Button>
    );
    const node = getByTestId("country");
    fireEvent.click(node);
});

afterEach(cleanup);
