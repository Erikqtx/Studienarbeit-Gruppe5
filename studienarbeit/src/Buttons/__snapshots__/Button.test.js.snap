// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches the snapshot 1`] = `
<div
  className="parentbox"
>
  <button
    className="country"
    data-testid="country"
    onClick={[Function]}
  >
    Country
  </button>
  <button
    className="state"
    onClick={[Function]}
  >
    State
  </button>
  <div />
  <button
    className="year1"
    onClick={[Function]}
  >
    2015
  </button>
  <button
    className="year2"
    onClick={[Function]}
  >
    2016
  </button>
</div>
`;
